src/sensor_mpu9250.c

```diff
@@ -203,18 +200,22 @@ mp9250_stop(struct mpu9250 *mp, uint8_t oid)
     i2c_write(mp->i2c->i2c_config, sizeof(msg), msg);
     uint32_t end2_time = timer_read_time();
 
-    // Drain any measurements still in fifo
-    uint16_t fifo_bytes = get_fifo_status(mp);
-    while (fifo_bytes >= BYTES_PER_FIFO_ENTRY) {
-        mp9250_query(mp, oid);
-        fifo_bytes = get_fifo_status(mp);
-    }
+    // Detect if a FIFO overrun occured
+    uint8_t int_reg[] = {AR_INT_STATUS};
+    uint8_t int_msg;
+    i2c_read(mp->i2c->i2c_config, sizeof(int_reg), int_reg, sizeof(int_msg),
+                &int_msg);
+    if (int_msg & FIFO_OVERFLOW_INT)
+        mp->limit_count++;
 
     // Report final data
     if (mp->data_count > 0)
         mp9250_report(mp, oid);
+    uint16_t bytes_to_read = get_fifo_status(mp);
     mp9250_status(mp, oid, end1_time, end2_time,
-                    fifo_bytes / BYTES_PER_FIFO_ENTRY);
+                    bytes_to_read / BYTES_PER_FIFO_ENTRY);
+    output("mp9250_stop limit_count=%hu fifo_max=%hu", mp->limit_count,
+            mp->fifo_max);
```

Comment:
As you've indicated, we'd want to remove the debugging code (and all references to `fifo_max`) before merging.